extra/mariadbfrm/mariadb_frm.cc

```diff
@@ -20,18 +20,79 @@
 
 #include "mariadb.h"
 
-#include "mysqld.h"
-#include "sql_class.h"
-#include "table.h"
-#include "sql_table.h"
-#include "sql_parse.h"
-#include "sql_plugin.h"
-#include <my_dir.h>
+#include "mariadb.h"
+#include "my_sys.h"
+#include "mysql_com.h"
+#include "m_ctype.h"
 #include "field.h"
 
 
+class Security_context;
+class LEX;
+class Transaction_state;
+
+class THD {
```

Comment:
No, You can't add THD like this. It'll be available only for mariadb_frm.cc. Other modules will not see it like this. Think of what will happen with access to some field if it has a different layout in  another module's function!